libs/constrained-generators/src/Constrained/Test.hs

```diff
@@ -547,3 +550,22 @@ parallelLetPair = constrained $ \p ->
       ]
   , match p $ \x y -> y <=. x
   ]
+
+andPair :: Spec Fn (Int, Int)
+andPair = constrained $ \p ->
+  match p $ \x y ->
+    and_ (x <=. 5) (y <=. 1)
+
+orPair :: Spec Fn (Int, Int)
+orPair = constrained $ \p ->
+  match p $ \x y ->
+    or_ (x <=. 5) (y <=. 5)
+
+data Three = One | Two | Three
+  deriving (Ord, Eq, Show, Generic)
+
+instance HasSimpleRep Three
+instance IsUniverse fn => HasSpec fn Three
+
+mapSizeConstrained :: Spec Fn (Map Three Int)
+mapSizeConstrained = constrained $ \m -> size_ (dom_ m) <=. 3
```

Comment:
Remove these from this PR as they are in another PR that shouldn't interfere with this.

Reply:
So I based this OR on   PR-map-size-test, so we would have this baseline test at the beginning of the 3 phase (PRS) that lead up to the whole shebang. Isn't that what you wanted?

Reply:
That works too.